tfep/io/log.py

```diff
@@ -534,16 +582,37 @@ def _build_mask(self, remove_nans, data_type):
 
         return mask
 
-    def _dump_data(self, data_type):
+    def _dump_data(self, data_type: str, state_mapping: Tuple[int, int]):
         """Dump on disk the currently loaded training data."""
         data_attr = '_loaded_' + data_type + '_data'  # e.g., _loaded_train_data
-        file_path = self._get_data_file_path(data_type=data_type)
-        np.savez_compressed(file_path, **getattr(self, data_attr))
-
-    def _get_data_file_path(self, data_type):
+        file_path = self._get_data_file_path(data_type, state_mapping)
+        
+        with h5py.File(file_path, 'w') as f:
```

Comment:
I'd not create a new file everytime. If we do, we might as well use the npz format. It's simpler and probably faster.